src/coffee/directives/api/utils/_async.coffee

```diff
@@ -137,6 +137,27 @@ angular.module('uiGmapgoogle-maps.directives.api.utils')
     true
 
 
+  _getIterateeValue = (collection, array, index) ->
+    _isArray = collection == array
```

Comment:
Get the value and exit out immediately if it is an array otherwise valOrKey is a key and use that to get the val from the object.